README.md

```diff
@@ -145,16 +177,17 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
 
+Wrote and added a file called "main.cpp"
```

Comment:
Thanks for writing code to exercise you function. However, they're not really tests. A test is code that knows what it expects, and that report when the actual result is different. There should be no human involvement, except perhaps getting notified that tests have failed.

# -2

And the we have the implementation. The idea of this exercise was to have you write a state machine, ideally a table driven one. That's why it asks you to "Translate this table into a programming language of your choice". 

You've written a manual parser that simply looks at words. There's no mention of states. 

Here's the kind of thing I was looking for:

~~~ cpp
#include <iostream>
using namespace std;
#include <map>

typedef int state;
typedef std::map<std::string, state> transitions;
typedef std::map<state, transitions> state_table;

typedef vector<string> sentence;


const int END = 99;

state_table states = {
// current    transitions
// state      {word -> new state}
// ---------+-------------------------------------------------
  {     0,    {{ "the",    1 }}},
	{     1,    {{ "lazy",   2 }, { "smelly",  2 },   { "dog", 3 }, {"cat", 3 }}},
	{     2,    {{ "lazy",   2 }, { "smelly",  2 },   { "dog", 3 }, {"cat", 3 }}},
	{     3,    {{ "ate",    4 }, { "ran",     4 }}},
	{     4,    {{ "slowly", 5 }, { "noisily", 5 },   { "EOI", END }}},
	{     5,    {{ "EOI",    END }}}
};


int getNextState(int currentState, string nextWord) {
	transitions t = states[currentState];
	if (t.count(nextWord))
		return t[nextWord];
	else
		return -1;
}

bool validSentence(sentence words) {
    int state = 0;
    for (string word: words) {
        state = getNextState(state, word);
        if (state < 0)
            return false;
    }
    return state == 5;
}
~~~

# -6